framework/behaviors/AttributeBehavior.php

```diff
@@ -118,6 +123,10 @@ public function evaluateAttributes($event)
             foreach ($attributes as $attribute) {
                 // ignore attribute names which are not string (e.g. when set by TimestampBehavior::updatedAtAttribute)
                 if (is_string($attribute)) {
+                    // skip attribute update when it is not empty and the corresponding mode is enabled
+                    if ($this->skipUpdateOnFilled && !empty($this->owner->$attribute)) {
```

Comment:
>  'insert' is not having existingValues right?
Insert does have `existing` values if user provided them. User provides values (they exist since this time), then the behavior performs the checks and evaluates them if needed.